interfaces/connection.go

```diff
@@ -57,6 +57,18 @@ type Attrer interface {
 	Lookup(path string) (value interface{}, ok bool)
 }
 
+type SnippetKey struct {
+	key string
+}
+
+func (pk *SnippetKey) GetValue() string {
```

Comment:
This could be just String
```suggestion
func (pk *SnippetKey) String() string {
```

Reply:
Done.